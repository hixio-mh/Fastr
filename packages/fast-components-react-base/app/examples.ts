import AutoSuggestExamples from "../src/auto-suggest/examples.data";
export { AutoSuggestExamples };

import BadgeExamples from "../src/badge/examples.data";
export { BadgeExamples };

import BreadcrumbExamples from "../src/breadcrumb/examples.data";
export { BreadcrumbExamples };

import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CardExamples from "../src/card/examples.data";
export { CardExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import ContextMenu from "../src/context-menu/examples.data";
export { ContextMenu };

import ContextMenuItem from "../src/context-menu-item/examples.data";
export { ContextMenuItem };

import DialogExamples from "../src/dialog/examples.data";
export { DialogExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import HorizontalOverflowExamples from "../src/horizontal-overflow/examples.data";
export { HorizontalOverflowExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ListboxExamples from "../src/listbox/examples.data";
export { ListboxExamples };

import ListboxItemExamples from "../src/listbox-item/examples.data";
export { ListboxItemExamples };

import NumberFieldExamples from "../src/number-field/examples.data";
export { NumberFieldExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import RadioExamples from "../src/radio/examples.data";
export { RadioExamples };

import SelectExamples from "../src/select/examples.data";
export { SelectExamples };

import SliderExamples from "../src/slider/examples.data";
export { SliderExamples };

import SliderTrackItemExamples from "../src/slider-track-item/examples.data";
export { SliderTrackItemExamples };

import TabsExamples from "../src/tabs/examples.data";
export { TabsExamples };

import TextAreaExamples from "../src/text-area/examples.data";
export { TextAreaExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
