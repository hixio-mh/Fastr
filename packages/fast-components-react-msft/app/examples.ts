import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CallToActionExamples from "../src/call-to-action/examples.data";
export { CallToActionExamples };

import CaptionExamples from "../src/caption/examples.data";
export { CaptionExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import Dialog from "../src/dialog/examples.data";
export { Dialog };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import MetatextExamples from "../src/metatext/examples.data";
export { MetatextExamples };

import ParagraphExamples from "../src/paragraph/examples.data";
export { ParagraphExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import SubheadingExamples from "../src/subheading/examples.data";
export { SubheadingExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
