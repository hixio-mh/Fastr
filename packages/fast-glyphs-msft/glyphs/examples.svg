<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>