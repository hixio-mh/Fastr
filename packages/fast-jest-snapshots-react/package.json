{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "sideEffects": false,
  "version": "2.3.0",
  "author": {
    "name": "Microsoft",
    "url": "https://discord.gg/FcSNfg4"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "build": "tsc -p ./tsconfig.json",
    "prepare": "npm run build",
    "start": "webpack-dev-server",
    "test": "npm run tslint",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'",
    "watch": "npm run build -- -w --preserveWatchOutput"
  },
  "jest": {
    "testURL": "http://localhost",
    "transform": {
      "^.+\\.tsx?$": "ts-jest",
      "^.+\\.jsx?$": "babel-jest"
    },
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx"
    ]
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^2.2.0",
    "@types/jest": "^22.2.2",
    "@types/react": "^16.3.6",
    "jest": "^22.4.3",
    "react": "^16.5.2",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^3.0.1"
  },
  "dependencies": {
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
