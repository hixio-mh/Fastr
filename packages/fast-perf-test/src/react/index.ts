import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
