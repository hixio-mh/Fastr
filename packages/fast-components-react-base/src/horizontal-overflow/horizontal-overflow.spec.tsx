import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, mount } from "enzyme";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import HorizontalOverflow, {
    ButtonDirection,
    IHorizontalOverflowClassNameContract
} from "./";
import examples from "./examples.data";
import "raf/polyfill";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

const imageSet1: JSX.Element[] = [
    (<img key="image1" src="https://placehold.it/200x200?text=1" />),
    (<img key="image2" src="https://placehold.it/200x200?text=2" />),
    (<img key="image3" src="https://placehold.it/200x200?text=3" />),
    (<img key="image4" src="https://placehold.it/200x200?text=4" />),
    (<img key="image5" src="https://placehold.it/200x200?text=5" />),
    (<img key="image6" src="https://placehold.it/200x200?text=6" />)
];

const managedClasses: IHorizontalOverflowClassNameContract = {
    horizontalOverflow: "horizontal-overflow-class",
    horizontalOverflow_contentRegion: "horizontal-overflow-items-class",
    horizontalOverflow_next: "horizontal-overflow-next-class",
    horizontalOverflow_previous: "horizontal-overflow-previous-class"
};

describe("horizontal overflow snapshot", (): void => {
    generateSnapshots(examples);
});

// TODO #746: https://github.com/Microsoft/fast-dna/issues/746

/* tslint:disable:no-string-literal */
describe("horizontal overflow", (): void => {
    test("should render a previous button if one is passed as a child with the appropriate slot prop", () => {
        const renderedWithImagesAndPrevious: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                <button id="testButtonPrevious" slot="previous">previous</button>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(renderedWithImagesAndPrevious.find("#testButtonPrevious")).not.toBe(undefined);
    });
    test("should render a next button if one is passed as a child with the appropriate slot prop", () => {
        const renderedWithImagesAndNext: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                <button id="testButtonNext" slot="next">next</button>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(renderedWithImagesAndNext.find("#testButtonNext")).not.toBe(undefined);
    });
    test("should render a series of items if they are passed as children", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(renderedWithImages.find("img").length).toBe(6);
    });
    test("should not set `itemHeight` if no children have been passed", () => {
        const rendered: any = mount(
            <HorizontalOverflow managedClasses={managedClasses} />
        );

        expect(rendered.state("itemsHeight")).toBe(0);
    });
    test("should update the rendering of a series of items if they are modified", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(renderedWithImages.find("img").length).toBe(6);

        renderedWithImages.setProps({ children: [
            (<img key="image1" src="https://placehold.it/200x200?text=1" />),
            (<img key="image2" src="https://placehold.it/200x200?text=2" />)
        ]});

        expect(renderedWithImages.find("img").length).toBe(2);
    });
    test("should update the scrolled distance when moving next to include the next number of items that can be in view", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImages.instance()[
                "getScrollDistanceFromDirection"
            ](
                ButtonDirection.next,
                50,
                [10, 20, 20, 50, 20],
                0
            )
        ).toBe(50);

        // reaches the max distance and uses that instead
        expect(
            renderedWithImages.instance()[
                "getScrollDistanceFromDirection"
            ](
                ButtonDirection.next,
                40,
                [10, 20, 10, 30, 20, 10],
                30
            )
        ).toBe(60);

        expect(
            renderedWithImages.instance()[
                "getScrollDistanceFromDirection"
            ](
                ButtonDirection.next,
                40,
                [10, 20, 10, 30, 20, 10, 20, 10],
                30
            )
        ).toBe(70);

        expect(
            renderedWithImages.instance()[
                "getScrollDistanceFromDirection"
            ](
                ButtonDirection.next,
                40,
                [10, 20, 10, 30, 20, 10, 20, 10],
                40
            )
        ).toBe(70);
    });
    test("should update the scrolled distance when moving previous to include the previous number of items that can be in view", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImages.instance()[
                "getScrollDistanceFromDirection"
            ](
                ButtonDirection.previous,
                50,
                [10, 20, 20, 50, 20],
                10
            )
        ).toBe(0);

        expect(
            renderedWithImages.instance()[
                "getScrollDistanceFromDirection"
            ](
                ButtonDirection.previous,
                50,
                [50, 50, 50, 50],
                100
            )
        ).toBe(50);
    });
    test("should an state property `itemsHeight`", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(renderedWithImages.state("itemsHeight")).toBe(0);
        expect(
            renderedWithImages.instance()[
                "getItemMaxHeight"
            ]()
        ).toBe(0);
    });
    test("should ease the animation correctly when moving the scroll position", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImages.instance()[
                "easeInOutQuad"
            ](
                1,
                0,
                0.5,
                50
            )
        ).toBe(0.0004);
    });
    test("should get the distance when moving next/previous", () => {
        const renderedWithImagesAndNextAndPrevious: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                <button id="testButtonNext" slot="next">next</button>
                <button id="testButtonPrevious" slot="previous">previous</button>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getNextDistance"
            ](500, [120, 140, 80, 220, 210, 100, 90, 200, 190, 170, 180, 210, 190], 0)
        ).toBe(340);

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getNextDistance"
            ](500, [120, 140, 80, 220, 210, 100, 90, 200, 190, 170, 180, 210, 190], 2100)
        ).toBe(2100);

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getPreviousDistance"
            ](500, [120, 140, 80, 220, 210, 100, 90, 200, 190, 170, 180, 210, 190], 560)
        ).toBe(120);

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getPreviousDistance"
            ](500, [120, 140, 80, 220, 210, 100, 90, 200, 190, 170, 180, 210, 190], 0)
        ).toBe(0);

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getScrollDistanceFromDirection"
            ]("next", 500, [], 0)
        ).toBe(0);
    });
    test("should set a max/min distance without additional calculations", () => {
        const renderedWithImagesAndNextAndPrevious: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                <button id="testButtonNext" slot="next">next</button>
                <button id="testButtonPrevious" slot="previous">previous</button>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getWithinMaxDistance"
            ](400, 500, [], 400)
        ).toBe(400);

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "getWithinMinDistance"
            ](0, 500, [])
        ).toBe(0);
    });
    test("should have an `onLoad` method", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow className="foo" managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImages.instance()[
                "itemsOnLoad"
            ]()
        ).toBe(undefined);

        expect(renderedWithImages.instance().state.itemsHeight).toBe(0);

        renderedWithImages.setState({ itemsHeight: 50});

        expect(renderedWithImages.instance().state.itemsHeight).toBe(50);

        renderedWithImages.find("div.foo").simulate("load", {});

        expect(renderedWithImages.instance().state.itemsHeight).toBe(0);
    });
    test("should execute a scroll animation on the element", () => {
        const renderedWithImages: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                {imageSet1}
            </HorizontalOverflow>
        );

        const itemsElement: HTMLDivElement = renderedWithImages.find(".horizontal-overflow-items-class");

        expect(
            renderedWithImages.instance()[
                "scrollLeft"
            ](
                itemsElement,
                50,
                0
            )
        ).toBe(undefined);
    });
    test("should allow clicks on previous and next buttons", () => {
        const renderedWithImagesAndNextAndPrevious: any = mount(
            <HorizontalOverflow managedClasses={managedClasses}>
                <button id="testButtonNext" slot="next">next</button>
                <button id="testButtonPrevious" slot="previous">previous</button>
                {imageSet1}
            </HorizontalOverflow>
        );

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "handleNextClick"
            ]()
        ).toBe(undefined);

        expect(
            renderedWithImagesAndNextAndPrevious.instance()[
                "handlePreviousClick"
            ]()
        ).toBe(undefined);
    });
});
/* tslint:enable:no-string-literal */
