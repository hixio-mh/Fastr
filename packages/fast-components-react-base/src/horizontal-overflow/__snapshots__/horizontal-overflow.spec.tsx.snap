// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`horizontal overflow snapshot Horizontal overflow: 0 1`] = `
<div
  className="horizontal-overflow"
  onLoad={[Function]}
>
  <div
    style={
      Object {
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <ul
      className="horizontal-overflow_items"
      style={
        Object {
          "margin": 0,
          "overflowX": "scroll",
          "padding": 0,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100/414141/?text=1"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/180x100?text=2"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100/414141/?text=3"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100?text=4"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/140x100/414141/?text=5"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/200x100?text=6"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/220x100/414141/?text=7"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/170x100?text=8"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/160x100/414141/?text=9"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100?text=10"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/110x100/414141/?text=11"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/270x100?text=12"
          srcSet={null}
        />
      </li>
    </ul>
  </div>
  <div
    className="horizontal-overflow_previous"
    onClick={[Function]}
  >
    <button
      className="button"
      href={null}
      slot="previous"
    >
      previous
    </button>
  </div>
  <div
    className="horizontal-overflow_next"
    onClick={[Function]}
  >
    <button
      className="button"
      href={null}
      slot="next"
    >
      next
    </button>
  </div>
</div>
`;

exports[`horizontal overflow snapshot Horizontal overflow: 1 1`] = `
<div
  className="horizontal-overflow"
  onLoad={[Function]}
>
  <div
    style={
      Object {
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <ul
      className="horizontal-overflow_items"
      style={
        Object {
          "margin": 0,
          "overflowX": "scroll",
          "padding": 0,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100/414141/?text=1"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/180x100?text=2"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100/414141/?text=3"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100?text=4"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/140x100/414141/?text=5"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/200x100?text=6"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/220x100/414141/?text=7"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/170x100?text=8"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/160x100/414141/?text=9"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100?text=10"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/110x100/414141/?text=11"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/270x100?text=12"
          srcSet={null}
        />
      </li>
    </ul>
  </div>
  <div
    className="horizontal-overflow_previous"
    onClick={[Function]}
  />
  <div
    className="horizontal-overflow_next"
    onClick={[Function]}
  >
    <button
      className="button"
      href={null}
      slot="next"
    >
      next
    </button>
  </div>
</div>
`;

exports[`horizontal overflow snapshot Horizontal overflow: 2 1`] = `
<div
  className="horizontal-overflow"
  onLoad={[Function]}
>
  <div
    style={
      Object {
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <ul
      className="horizontal-overflow_items"
      style={
        Object {
          "margin": 0,
          "overflowX": "scroll",
          "padding": 0,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100/414141/?text=1"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/180x100?text=2"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100/414141/?text=3"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100?text=4"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/140x100/414141/?text=5"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/200x100?text=6"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/220x100/414141/?text=7"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/170x100?text=8"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/160x100/414141/?text=9"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100?text=10"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/110x100/414141/?text=11"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/270x100?text=12"
          srcSet={null}
        />
      </li>
    </ul>
  </div>
  <div
    className="horizontal-overflow_previous"
    onClick={[Function]}
  >
    <button
      className="button"
      href={null}
      slot="previous"
    >
      previous
    </button>
  </div>
  <div
    className="horizontal-overflow_next"
    onClick={[Function]}
  />
</div>
`;

exports[`horizontal overflow snapshot Horizontal overflow: 3 1`] = `
<div
  className="horizontal-overflow"
  onLoad={[Function]}
>
  <div
    style={
      Object {
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <ul
      className="horizontal-overflow_items"
      style={
        Object {
          "margin": 0,
          "overflowX": "scroll",
          "padding": 0,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100/414141/?text=1"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/180x100?text=2"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100/414141/?text=3"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/120x100?text=4"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/140x100/414141/?text=5"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/200x100?text=6"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/220x100/414141/?text=7"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/170x100?text=8"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/160x100/414141/?text=9"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/240x100?text=10"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/110x100/414141/?text=11"
          srcSet={null}
        />
      </li>
      <li
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt="placeholder image"
          className="image"
          sizes={null}
          src="https://placehold.it/270x100?text=12"
          srcSet={null}
        />
      </li>
    </ul>
  </div>
  <div
    className="horizontal-overflow_previous"
    onClick={[Function]}
  />
  <div
    className="horizontal-overflow_next"
    onClick={[Function]}
  />
</div>
`;

exports[`horizontal overflow snapshot Horizontal overflow: 4 1`] = `
<div
  className="horizontal-overflow"
  onLoad={[Function]}
>
  <div
    style={
      Object {
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <ul
      className="horizontal-overflow_items"
      style={
        Object {
          "margin": 0,
          "overflowX": "scroll",
          "padding": 0,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    />
  </div>
  <div
    className="horizontal-overflow_previous"
    onClick={[Function]}
  />
  <div
    className="horizontal-overflow_next"
    onClick={[Function]}
  />
</div>
`;
